packages/stage-ui/src/components/FirstTimeSetup/FirstTimeSetupDialog.vue

```diff
@@ -0,0 +1,365 @@
+
+    // Validate using provider's validator
+    const metadata = providersStore.getProviderMetadata(selectedProvider.value.id)
+    isValid.value = await metadata.validators.validateProviderConfig(config)
+
+    if (isValid.value) {
+      validationMessage.value = t('firstTimeSetup.validationSuccess')
+    }
+    else {
+      validationMessage.value = t('firstTimeSetup.validationFailed')
+    }
+  }
+  catch (error) {
+    isValid.value = false
+    validationMessage.value = t('validationError', {
+      error: error instanceof Error ? error.message : String(error),
+    })
+  }
+  finally {
+    isValidating.value = false
+  }
+}
+
+// Debounced validation function
+const debouncedValidateConfiguration = useDebounceFn(() => {
+  if (!selectedProvider.value)
+    return
+  if (needsApiKey.value && !apiKey.value.trim())
+    return
+  if (needsBaseUrl.value && !baseUrl.value.trim())
+    return
+  if (selectedProvider.value.id === 'cloudflare-workers-ai' && !accountId.value.trim())
+    return
+
+  validateConfiguration()
+}, 500)
+
+// Watch for changes and validate
+watch([apiKey, baseUrl, accountId], () => {
+  if (selectedProvider.value && (apiKey.value || baseUrl.value || accountId.value)) {
+    debouncedValidateConfiguration()
+  }
+}, { deep: true })
+
+// Actions
+function handleSkip() {
+  showDialog.value = false
+  emit('skipped')
+}
+
+async function handleSave() {
+  if (!selectedProvider.value || !canSave.value)
+    return
+
+  // Save configuration to providers store
+  const config: Record<string, unknown> = {}
+
+  if (needsApiKey.value)
+    config.apiKey = apiKey.value.trim()
+  if (needsBaseUrl.value)
+    config.baseUrl = baseUrl.value.trim()
+  if (selectedProvider.value.id === 'cloudflare-workers-ai')
+    config.accountId = accountId.value.trim()
+
+  providers.value[selectedProvider.value.id] = {
+    ...providers.value[selectedProvider.value.id],
+    ...config,
+  }
+
+  showDialog.value = false
+  emit('configured')
+}
+
+// Initialize with first popular provider
+onMounted(() => {
+  if (popularProviders.value.length > 0) {
+    selectedProviderId.value = popularProviders.value[0].id
+    selectProvider(popularProviders.value[0])
+  }
+})
+</script>
+
+<template>
+  <DialogRoot :open="showDialog" @update:open="value => showDialog = value">
+    <DialogPortal>
+      <DialogOverlay class="data-[state=open]:animate-fadeIn data-[state=closed]:animate-fadeOut fixed inset-0 z-[9999] bg-black/50 backdrop-blur-sm" />
+      <DialogContent class="data-[state=open]:animate-contentShow data-[state=closed]:animate-contentHide fixed left-1/2 top-1/2 z-[9999] mx-4 max-w-2xl w-[92vw] transform rounded-lg bg-white p-8 shadow-xl -translate-x-1/2 -translate-y-1/2 dark:bg-neutral-900">
+        <!-- Header -->
+        <div class="mb-8 text-center">
+          <div class="mb-4 flex justify-center">
+            <div class="rounded-full from-blue-500 to-purple-600 bg-gradient-to-br p-4">
```

Comment:
```suggestion
            <div class="rounded-full from-primary to-primary-600 bg-gradient-to-br p-4">
```

I would recommend using primary color for this.